v2/internal/controllers/controller_resources.go

```diff
@@ -52,16 +53,31 @@ func CreateScheme() *runtime.Scheme {
 	return scheme
 }
 
-//GetResourceExtensions returns a map between resource and resource extension
-func GetResourceExtensions() map[schema.GroupVersionKind]genruntime.ResourceExtension {
+// GetResourceExtensions returns a map between resource and resource extension
+func GetResourceExtensions(scheme *runtime.Scheme) map[schema.GroupVersionKind]genruntime.ResourceExtension {
 
+	var log logr.Logger
 	extensionMapping := make(map[schema.GroupVersionKind]genruntime.ResourceExtension)
 
 	for _, extension := range getResourceExtensions() {
 		for _, resource := range extension.GetExtendedResources() {
-			extensionMapping[resource.GroupVersionKind()] = extension
+
+			// Make sure the type casting goes well, and we can extract the GVK successfully.
+			resourceObj, ok := resource.(runtime.Object)
+			if !ok {
+				log.V(Status).Info("Unexpected resource type", "resource", resource.AzureName(), "actual", fmt.Sprintf("%T", resource))
```

Comment:
Instead of logging here, just have this return `(map[schema.GroupVersionKind]genruntime.ResourceExtension, error)`.

Instead of logging, use `errors.Errorf` and create an error with the same message you were logging here. This gives the caller the opportunity to choose the logger to direct this to, which we need because these methods are called in different contexts (both in our tests w/ the test logger and in our actual controller with its logger).